app/controllers/SubmissionsController.scala

```diff
@@ -86,8 +89,15 @@ class SubmissionsController @Inject()(
     }
   }
 
-  def viewDeclaration(id: String): Action[AnyContent] = (authenticate andThen verifyEmail) { implicit request =>
-    Redirect(controllers.declaration.routes.SubmittedDeclarationController.displayPage()).addingToSession(ExportsSessionKeys.declarationId -> id)
+  def viewDeclaration(id: String): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
+    customsDeclareExportsConnector.findDeclaration(id).flatMap {
+      case Some(declaration) =>
+        customsDeclareExportsConnector.findNotifications(id).map { notifications =>
+          Ok(submittedDeclarationPage(notifications, declaration))
```

Comment:
Could we clear the `declarationId` session val here to avoid the confusion of having a different Id in the cache than the dec you are viewing?

Reply:
I've decided to clear this from session on Choice page for the time being. We'll see if we get a decision to move away from storing this data in session and then clearing will not be required anymore.